realtime/utils.js

```diff
@@ -400,7 +401,15 @@ function isIpWhitelisted(addr, whitelist) {
  */
 function attachAspectSubject(sample, useSampleStore, subjectModel,
   aspectModel) {
-  const nameParts = sample.name.split('|');
+  let nameParts;
+
+  // check if sample object contains name
+  if (sample.name) {
+    nameParts = sample.name.split('|');
+  } else {
+    logger.error('sample object does not contain name', sample);
+  }
```

Comment:
we should not attempt to continue here if sample is falsey OR sample does contain "name" attribute OR sample.name does not contain "|" character, AND we should go back to caller and try to figure out under what circumstances we're passing an invalid sample through (more logging there?)